Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/conflict.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2020-2024 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+
+from .command import Command
+from webkitcorepy import run
+
+
+class Conflict(Command):
+    name = 'conflict'
+    help = "Given the representative Radar ID of a conflicting merge, checkout the branch and setup the integration remote if needed."
+
+    INTEGRATION_REMOTE_NAME = "integration"
+    INTEGRATION_REMOTE_URL = "git@github.com:apple/WebKit-integration.git"
+
+    @classmethod
+    def parser(cls, parser, loggers=None):
+        parser.add_argument(
+            'radar_id',
+            type=int, default=None,
+            help='Radar ID that caused a merge conflict.',
+        )
+
+    @classmethod
+    def ensure_integration_remote_setup(cls, repository):
+        local_remotes = run(
+            [repository.executable(), 'remote'], cwd=repository.root_path, capture_output=True,
+        ).stdout.decode('utf-8').strip().splitlines()
+        if cls.INTEGRATION_REMOTE_NAME in local_remotes:
+            return True
+
+        sys.stdout.write('Adding local remote {} with url {}'.format(cls.INTEGRATION_REMOTE_NAME, cls.INTEGRATION_REMOTE_URL))
+        return run(
+            [repository.executable(), 'remote', 'add', cls.INTEGRATION_REMOTE_NAME, cls.INTEGRATION_REMOTE_URL],
+            cwd=repository.root_path,
+            capture_output=False,
+        ).returncode == 0
+
+    @classmethod
+    def main(cls, args, repository, **kwargs):
+        if not repository:
+            sys.stderr.write('No repository provided\n')
+            return 1
+        if not repository.path:
+            sys.stderr.write('Cannot checkout conflict on remote repository\n')
+            return 1
```

Comment:
There should be another check, right? As this logic only applies to a `git` checkout?